src/plugins/core/borders.ts

```diff
@@ -90,9 +90,55 @@ export class BordersPlugin extends CorePlugin<BordersPluginState> implements Bor
         const elements = [...cmd.elements].sort((a, b) => b - a);
         for (const group of groupConsecutive(elements)) {
           if (cmd.dimension === "COL") {
-            this.shiftBordersHorizontally(cmd.sheetId, group[group.length - 1] + 1, -group.length);
+            if (group[0] == this.getters.getNumberCols(cmd.sheetId)) {
+              for (let row: HeaderIndex = 0; row < this.getters.getNumberRows(cmd.sheetId); row++) {
```

Comment:
That doesn't work if we remove the two last columns of the spreadsheet 😔 It seems a bit annoying since you compare the columns before they are removed from the spreadsheet (groups) and the numbre of columns of the sheet after the removal (`getNumberCols`, since the sheet plugin is already updated when you are here)

Same thing for the first cols and the rows